src/execution/execute.ts

```diff
@@ -431,7 +432,9 @@ function executeFields(
   const results = Object.create(null);
   let containsPromise = false;
 
-  for (const [responseName, fieldNodes] of fields.entries()) {
+  const fieldsEntries = Array.from(fields.entries());
+  for (let i = 0; i < fieldsEntries.length; i++) {
+    const { 0: responseName, 1: fieldNodes } = fieldsEntries[i];
```

Comment:
```suggestion
    const [responseName, fieldNodes] = fieldsEntries[i];
```
This version is less wierd

Reply:
That version invokes the iterator and is very slow though